shorebird_code_push/lib/src/shorebird_code_push.dart

```diff
@@ -98,28 +92,19 @@ class ShorebirdCodePush {
     return nextPatch != null && currentPatch != nextPatch;
   }
 
-  void _handleError(Object error) {
-    final logMessage = '$_loggingPrefix $error';
-    if (error is ArgumentError) {
-      // ffi function lookup failures manifest as ArgumentErrors.
-      throw ShorebirdCodePushNotAvailableException();
-    } else {
-      throw ShorebirdCodePushException(logMessage);
-    }
-  }
-
-  /// Creates an [Updater] in a separate isolate and runs the given function. If
-  /// an error occurs, the error is logged and [fallbackValue] is returned.
-  Future<T> _runInIsolate<T>(
-    T Function(Updater updater) f, {
-    required T fallbackValue,
-  }) async {
+  /// Creates an [Updater] in a separate isolate and runs the given function.
+  Future<T> _runInIsolate<T>(T Function(Updater updater) f) async {
     try {
       // Create a new Updater in the new isolate.
       return await Isolate.run(() => f(_buildUpdater()));
     } catch (error) {
-      _handleError(error);
-      return fallbackValue;
+      final logMessage = '$_loggingPrefix $error';
+      if (error is ArgumentError) {
+        // ffi function lookup failures manifest as ArgumentErrors.
+        throw ShorebirdCodePushNotAvailableException();
```

Comment:
I'm skeptical that we want code to run in `shorebird run` and then fail in `flutter run` by default.  It goes against the sell proposition that shorebird is "drop in".  I'm open to it, but I'm still not convinced.  Would love to learn more on monday.

Reply:
I'm not 100% sold on it either - I'll think about it some more over the weekend. This is largely a response to people wanting more info from the errors (see the PR this is based on) and a helping a lot of people with "I made a patch, why isn't it downloading" issues this week.

Reply:
Thought about this some more and came up with (what I think is) a better idea. I've made the `ShorebirdCodePush` constructor private and introduced a static `initialize` function that returns a non-null `ShorebirdCodePush` if our updater bindings are present or `null` if they are not.